src/routes/blog/post/how-to-set-up-sign-in-with-apple/+page.markdoc

```diff
@@ -0,0 +1,439 @@
+
+## 3.2 Registering an App ID
+
+An **App ID** is required for every app that uses Apple's services, including authentication.
+
+1. Go to **Certificates, Identifiers & Profiles** > **Identifiers**.
+2. Click the plus icon (+) beside **Identifiers** to create a new identifier, then select **App IDs**.
+3. Provide a **Description** and a unique **Bundle ID** (e.g., `com.example.myapp`).
+4. Under **Capabilities**, enable **Sign in with Apple**.
+5. Review the settings and click **Register**.
+
+## 3.3 Creating a Service ID for web authentication
+
+If you're building a web app, you'll need to create a **Service ID** to handle Apple Sign-In.
+
+1. In the **Identifiers** section, click the plus icon (+) and select **Service IDs**.
+2. Provide a **Description** and a unique **Identifier** (e.g., `com.example.myapp.web`).
+3. Click **Register** to create the Service ID.
+
+With your service ID created, you can configure it for **Sign in with Apple**. To do this:
+
+1. Click on the Service ID you created.
+2. Beside **Sign in with Apple**, tick the checkbox to enable it.
+3. Click **Configure** to set up the service.
+4. Select the **Primary App ID** you created earlier.
+5. Under **Register Website URLs**, enter the domain of your web app and the **Return URI** you got from Appwrite earlier.
+6. Click **Save** to apply the changes.
+
+This Service ID allows users to authenticate with Apple via your web application.
+
+## 3.4 Generating an Apple authentication key
+
+To securely communicate with Apple's servers, you need to generate an authentication key.
+
+1. Go to **Keys** and click **Create a Key**.
+2. Set a **Key Name** (e.g., "Apple Sign-In Key").
+3. Enable **Sign in with Apple** and click **Configure**.
+4. Select the **App ID** you created earlier.
+5. After saving the key, click **Register** and a new page will display the key details.
+6. Click **Download** to save the key as a `.p8` file.
+
+**Important**: Store the `.p8` file securely, as it can only be downloaded once.
+
+With these configurations, you can now go back to Appwrite and fill in the Apple credentials as described in Step 2.
+
+# Step 4: Implementing sign in with Apple in your project
+
+With your Apple Developer credentials and Appwrite OAuth2 settings configured, it's time to implement **Sign in with Apple** in your app. We'll walk through this step-by-step for each platform.
+
+## 4.1 Understanding the OAuth2 flow with Appwrite
+
+Before diving into platform-specific code, let's break down the OAuth2 flow in Appwrite. Here's how the process works:
+
+1. **Initialization**: Your app uses Appwrite's SDK to initiate an OAuth2 session by redirecting the user to Apple's login page.
+2. **Authentication**: The user authenticates with their Apple ID credentials on Apple's login page.
+3. **Token Exchange**: Once authenticated, Apple redirects the user back to your app with an authorization code.
```

Comment:
We use lower case for all titles, please double check that all others are also lower case

Reply:
Fixed